src/compas/data/data.py

```diff
@@ -350,3 +360,34 @@ def validate_json(self):
             print("Validation against the JSON schema of this object failed.")
             raise e
         return jsonstring
+
+    def hash(self, as_string=False):
```

Comment:
This is too close to the built-in `__hash__` magic method, but has different semantics, I would suggest a different name, maybe `sha256hash` or `unique_hash`, or `fingerprint`, or something else

Reply:
I vote for `sha256` or `fingerprint`